configure

```diff
@@ -174,6 +174,19 @@ while true; do
 
   if [[ -z "$TF_CUDNN_VERSION" ]]; then
     TF_CUDNN_EXT=""
+    # Resolve to the SONAME of the symlink.  Use readlink without -f
+    # to resolve exactly once to the SONAME.  E.g, libcudnn.so ->
+    # libcudnn.so.4
+    REALVAL=`readlink ${CUDNN_INSTALL_PATH}/lib64/libcudnn.so`
```

Comment:
Because the SONAME implies the binary compatibility -- if a user has 4.0.7 on their system and we build with 4.0.4, it should be compatible -- that's the ABI compatibility.  This is going off of what @3XX0 said https://github.com/tensorflow/tensorflow/issues/2525#issuecomment-223523008 there

I'm not entirely sure what I'm supposed to do for OS X -- @yaroslavvb do you know what the dylib's look like?